library/system/src/lib/yast2/execute.rb

```diff
@@ -37,76 +37,120 @@ class Execute
-    # @param args see http://www.rubydoc.info/github/openSUSE/cheetah/Cheetah.run
-    # @raise Cheetah::ExecutionFailed if the command fails
-    def self.on_target!(*args)
-      root = Yast::WFM.scr_root
-
-      if args.last.is_a? ::Hash
-        args.last[:chroot] = root
-      else
-        args.push(chroot: root)
+        args.none? ? object : object.run
       end
 
-      Cheetah.run(*args)
+      # Runs without chroot; a failure becomes a popup.
+      # Runs a command described by *args*,
+      # *disregarding* a `chroot(2)` specified by the installation (WFM.scr_root).
+      # Shows a {ReportClass#Error popup} if the command fails
+      # and returns `nil` in such case.
+      # It also globally switches the default Cheetah logger to
+      # {http://www.rubydoc.info/github/yast/yast-ruby-bindings/Yast%2FLogger Y2Logger}.
+      # @param args see http://www.rubydoc.info/github/openSUSE/cheetah/Cheetah.run
+      def locally(*args)
+        popup_error { locally!(*args) }
+      end
+
+      # Runs without chroot; a failure becomes an exception.
+      # Runs a command described by *args*,
+      # *disregarding* a `chroot(2)` specified by the installation (WFM.scr_root).
+      # It also globally switches the default Cheetah logger to
+      # {http://www.rubydoc.info/github/yast/yast-ruby-bindings/Yast%2FLogger Y2Logger}.
+      # @param args see http://www.rubydoc.info/github/openSUSE/cheetah/Cheetah.run
+      # @raise Cheetah::ExecutionFailed if the command fails
+      def locally!(*args)
+        object = Execute.new(args)
+
+        args.none? ? object : object.run
+      end
+
+      def stdout(*args)
+        Execute.new.stdout(*args)
+      end
+
+    private
+
+      def popup_error(&block)
+        block.call
+      rescue Cheetah::ExecutionFailed => e
+        Yast.import "Report"
+
+        Yast::Report.Error(
+          _(
+            "Execution of command \"%{command}\" failed.\n"\
+            "Exit code: %{exitcode}\n"\
+            "Error output: %{stderr}"
+          ) % {
+            command:  e.commands.inspect,
+            exitcode: e.status.exitstatus,
+            stderr:   e.stderr
+          }
+        )
+      end
+    end
+
+    def initialize(args = [], options = {})
+      @args = args
+      @options = options
     end
 
-    # Runs without chroot; a failure becomes a popup.
-    # Runs a command described by *args*,
-    # *disregarding* a `chroot(2)` specified by the installation (WFM.scr_root).
-    # Shows a {ReportClass#Error popup} if the command fails
-    # and returns `nil` in such case.
-    # It also globally switches the default Cheetah logger to
-    # {http://www.rubydoc.info/github/yast/yast-ruby-bindings/Yast%2FLogger Y2Logger}.
-    # @param args see http://www.rubydoc.info/github/openSUSE/cheetah/Cheetah.run
-    def self.locally(*args)
-      popup_error { locally!(*args) }
+    def stdout(*args)
+      self.args = args
+
+      add_options(stdout: :capture)
+
+      run
```

Comment:
just note, maybe it makes sense to also act same as others. so `args.empty? ? self : run`